wyzesense2mqtt.py

```diff
@@ -12,169 +10,266 @@
+    global SENSORS
+    SENSORS[sensor_mac] = dict()
+    SENSORS[sensor_mac]['name'] = "Wyze Sense {0}".format(sensor_mac)
+    SENSORS[sensor_mac]['class'] = ("motion" if sensor_type == "motion" else "opening")
+    if sensor_version is not None:
+        SENSORS[sensor_mac]['sw_version'] = sensor_version
+
+    LOGGER.info("Writing Sensors Config File")
+    write_yaml_file(SENSORS_CONFIG_FILE, SENSORS)
+
+# Send discovery topics
+def send_discovery_topics(sensor_mac):
+    global SENSORS, CONFIG
+    LOGGER.info("Publishing discovery topics for {0}".format(sensor_mac))
+
+    sensor_name = SENSORS[sensor_mac]['name']
+    sensor_class = SENSORS[sensor_mac]['class']
+    if SENSORS[sensor_mac].get('sw_version') is not None:
+        sensor_version = SENSORS[sensor_mac]['sw_version']
+    else:
+        sensor_version = ""
 
     device_payload = {
         'identifiers': ["wyzesense_{0}".format(sensor_mac), sensor_mac],
         'manufacturer': "Wyze",
-        'model': ("Motion Sensor" if sensor_type == "motion" else "Contact Sensor"),
-        'name': ("Wyze Sense Motion Sensor" if sensor_type == "motion" else "Wyze Sense Contact Sensor")
+        'model': ("Sense Motion Sensor" if sensor_class == "motion" else "Sense Contact Sensor"),
+        'name': sensor_name,
+        'sw_version': sensor_version
     }
 
-    device_classes = {
+    entity_payloads = {
         'state': {
-            'name': "Wyze Sense {0}".format(sensor_mac),
-            'dev_cla': ("motion" if sensor_type == "motion" else "opening"),
+            'name': sensor_name,
+            'dev_cla': sensor_class,
             'pl_on': "1",
             'pl_off': "0",
-            'json_attr_t': WYZESENSE2MQTT_TOPIC_ROOT + sensor_mac
+            'json_attr_t': CONFIG['wyzesense2mqtt_topic_root'] + sensor_mac
         },
         'signal_strength': {
-            'name': "Wyze Sense {0} Signal Strength".format(sensor_mac),
+            'name': "{0} Signal Strength".format(sensor_name),
             'dev_cla': "signal_strength",
             'unit_of_meas': "dBm"
         },
         'battery': {
-            'name': "Wyze Sense {0} Battery".format(sensor_mac),
+            'name': "{0} Battery".format(sensor_name),
             'dev_cla': "battery",
             'unit_of_meas': "%"
         }
     }
 
-    # Send Discovery Topics
-    for device_class in device_classes :
-        device_classes[device_class]['val_tpl'] = "{{{{ value_json.{0} }}}}".format(device_class)
-        device_classes[device_class]['uniq_id'] = "wyzesense_{0}_{1}".format(sensor_mac, device_class)
-        device_classes[device_class]['stat_t'] = WYZESENSE2MQTT_TOPIC_ROOT + sensor_mac
-        device_classes[device_class]['dev'] = device_payload
-        sensor_type = ("binary_sensor" if device_class == "state" else "sensor")
+    for entity in entity_payloads:
+        entity_payloads[entity]['val_tpl'] = "{{{{ value_json.{0} }}}}".format(entity)
```

Comment:
I'm no python expert, but there has to be a better way of doing this rather than all these nested braces.

Reply:
I seem to recall being unable to find anything else. Since Python also uses braces for magic, I needed to effectively escape the ones I needed to pass through to MQTT. There might be some other function I could call, and worth looking into, since it is mildly confusing to look at.

Reply:
So far I'm unable to find a better way to do this and pass the correct Jinja2 syntax to Hass in the discovery payload. If you think of anything, completely open to it being slightly more readable, but since I need the variable substitution, it seems like there just isn't a good alternative.